src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,92 @@
+
+@Tag(name = "RecommendationRequest")
+@RequestMapping("/api/recommendationrequest")
+@RestController
+public class RecommendationRequestController extends ApiController {
+    @Autowired
+    RecommendationRequestRepository recommendationRequestRepository;
+
+    /**
+     * This method returns a list of all recommendationrequests.
+     * @return a list of all recommendationrequests
+     */
+    @Operation(summary = "List all recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<RecommendationRequest> allRecommendationRequests() {
+        Iterable<RecommendationRequest> recommendationRequests = recommendationRequestRepository.findAll();
+        return recommendationRequests;
+    }
+
+    /**
+     * This method returns a single recommendationrequests.
+     * @param id id of the recommendationrequests to get
+     * @return a single recommendationrequests
+     */
+    @Operation(summary = "Get a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public RecommendationRequest getById(
+            @Parameter(name = "id") @RequestParam Long id) {
+            RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        return recommendationRequest;
+    }
+
+    /**
+     * This method creates a new recommendationrequests. Accessible only to users with the role "ROLE_ADMIN" until our roles can be configured.
+     * @param professorName professor name of request
+     * @param professorEmail professor email of request
+     * @param recommendationTypes recommendation types of request
+     * @param details details of request
+     * @param submissionDate submission date of request
+     * @return the save recommendationrequests (with it's id field set by the database)
+     */
+    @Operation(summary = "Create a new recommendationrequests")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public RecommendationRequest postRecommendationRequests(
+            @Parameter(name = "professorName") @RequestParam String professorName,
+            @Parameter(name = "professorEmail") @RequestParam String professorEmail,
+            @Parameter(name = "recommendationTypes") @RequestParam String recommendationTypes,
+            @Parameter(name = "details") @RequestParam String details,
+            @Parameter(name = "submissionDate") @RequestParam LocalDateTime submissionDate)
+            {
+        //get current date right now and set status to pending
+        CurrentUser currentUser = getCurrentUser();
+        RecommendationRequest recommendationRequest = new RecommendationRequest();
+
+        recommendationRequest.setProfessorName(professorName);
+        recommendationRequest.setProfessorEmail(professorEmail);
```

Comment:
```suggestion
        // code goes here that validates whether the professorId is valid, i.e
        // exists in the Users table, and the professor flag is true for that user
        recommendationRequest.setProfessorId(professorId);
```

Reply:
I am currently just setting the actual Professor by Autowiring the userReposutory and getting the User from the repository by the professor_id that is given - is that good? I didn't make a professor_id part inside my Entity, just a User professor